cmake/extensions.cmake

```diff
@@ -2990,6 +2990,77 @@ function(dt_chosen var)
   endif()
 endfunction()
 
+# Internal helper. Canonicalizes a path 'path' into the output
+# variable 'var'. This resolves aliases, if any. Child nodes may be
+# accessed via alias as well. 'var' is left undefined if the path does
+# not refer to an existing node.
+#
+# Example devicetree:
+#
+#   / {
+#           foo {
+#                   my-label: bar {
+#                           baz {};
+#                   };
+#           };
+#           aliases {
+#                   my-alias = &my-label;
+#           };
+#   };
+#
+# Example usage:
+#
+#   dt_path_internal(ret "/foo/bar")     # sets ret to "/foo/bar"
+#   dt_path_internal(ret "my-alias")     # sets ret to "/foo/bar"
+#   dt_path_internal(ret "my-alias/baz") # sets ret to "/foo/bar/baz"
+#   dt_path_internal(ret "/blub")        # ret is undefined
+function(dt_path_internal var path)
+  string(FIND "${path}" "/" slash_index)
```

Comment:
an alternative would be to use a `REGEX MATCH` like this:
```
string(REGEX MATCH "^([/])?([^/]+)(/.*)*" match_result "${PATH}")
```

For example that will give:
```
path: /foo                                                                                                                                                                                                                                   
CMAKE_MATCH_1: /                                                                                                                                                                                                                             
CMAKE_MATCH_2: foo                                                                                                                                                                                                                           
CMAKE_MATCH_3:                                                                                                                                                                                                                               

path: /foo/bar                                                                                                                                                                                                                               
CMAKE_MATCH_1: /                                                                                                                                                                                                                             
CMAKE_MATCH_2: foo                                                                                                                                                                                                                           
CMAKE_MATCH_3: /bar                                                                                                                                                                                                                          

path: my-alias                                                                                                                                                                                                                               
CMAKE_MATCH_1:                                                                                                                                                                                                                               
CMAKE_MATCH_2: my-alias                                                                                                                                                                                                                      
CMAKE_MATCH_3:                                                                                                                                                                                                                               

path: my-alias/baz                                                                                                                                                                                                                           
CMAKE_MATCH_1:                                                                                                                                                                                                                               
CMAKE_MATCH_2: my-alias                                                                                                                                                                                                                      
CMAKE_MATCH_3: /baz      
```

So you can use `CMAKE_MATCH_1` to identify if `path` starts with `/`.
`CMAKE_MATCH_2` as `alias_name` and if `CMAKE_MATCH_3` is defined, then it's an alias path.

Doing so could cleanup some logic below I believe.

Reply:
> an alternative would be to use a REGEX MATCH like this:

That regular expression doesn't work for the root node, `/`. I would prefer to leave this as-is.

Reply:
> That regular expression doesn't work for the root node, `/`.

I wasn't aware the root node was a use-case also.
In that case the regex should look as
```
string(REGEX MATCH "^([/])?([^/]*)(/.*)*" match_result "/") 
```

> I would prefer to leave this as-is.

I'm fine with that as well.

Reply:
>> I would prefer to leave this as-is.
>
> I'm fine with that as well.

OK, that was your last unresolved comment, then, I think. OK to approve?